core/domain/beam_job_domain.py

```diff
@@ -0,0 +1,191 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects related to Apache Beam jobs."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import python_utils
+import utils
+
+(beam_job_models,) = models.Registry.import_models([models.NAMES.beam_job])
+
+
+class BeamJob(python_utils.OBJECT):
+    """Encapsulates the definition of an Apache Beam job.
+
+    Attributes:
+        name: str. The name of the class that implements the job's logic.
+        parameter_names: list(str). The name of the parameters the job accepts
```

Comment:
Perhaps argument_names, since "parameter" is already a term of art in the codebase? http://oppia.github.io/#/Parameters